pkg/backup/restore/restore_manager.go

```diff
@@ -268,6 +268,63 @@ func (rm *restoreManager) validateRestore(r *v1alpha1.Restore, tc *v1alpha1.Tidb
 		}
 	}
 
+	// check tikv encrypt config
+	if err = rm.checkTiKVEncryption(r, tc); err != nil {
+		return fmt.Errorf("TiKV encryption missmatched with backup with error %v", err)
+	}
+	return nil
+}
+
+// volume snapshot restore support
+//
+//	both backup and restore with the same encryption
+//	backup without encryption and restore has encryption
+//
+// volume snapshot restore does not support
+//
+//	backup has encryption and restore has not
+func (rm *restoreManager) checkTiKVEncryption(r *v1alpha1.Restore, tc *v1alpha1.TidbCluster) error {
+	backupConfig, reason, err := rm.readTiKVConfigFromBackupMeta(r)
+	if err != nil {
+		klog.Errorf("read tiflash replica failure with reason %s", reason)
+		return err
+	}
+
+	// check encryption is enabled
+	if v := backupConfig.Get("security.encryption.data-encryption-method"); v != nil {
```

Comment:
if `v == nil`, it can indicate that there is no encryption in backup. I think we can `return nil` directly.

Reply:
https://github.com/pingcap/tidb-operator/pull/4914/commits/6a26fcafb80564c25d3d12e057ae19d0100ebe51